server/api/rounds.py

```diff
@@ -709,6 +709,55 @@ def sample_batches(
 def validate_round(round: dict, election: Election):
     validate(round, CREATE_ROUND_REQUEST_SCHEMA)
 
+    total_batches = 0
+    if election.audit_type == AuditType.BATCH_COMPARISON:
+        contest_sample_size_arr = list(round["sampleSizes"].values())
+        contest_sample_size = contest_sample_size_arr[0]
+        for single_contest in election.contests:
+            if single_contest.is_targeted:
+                for jurisdiction in single_contest.jurisdictions:
+                    total_batches += (
+                        jurisdiction.manifest_num_batches
+                        if jurisdiction.manifest_num_batches
+                        else 0
+                    )
+                if contest_sample_size > total_batches:
+                    raise Conflict(
+                        f"Sample size must be less than or equal to: {total_batches} (the total number of batches in the targeted contest)"
+                    )
+
+    count = 0
+    if election.audit_type == AuditType.BALLOT_POLLING:
+        for single_contest in election.contests:
+            if single_contest.is_targeted:
+                contest_sample_size_arr = list(round["sampleSizes"].values())
+                contest_sample_size = contest_sample_size_arr[count]
+                if contest_sample_size > single_contest.total_ballots_cast:
+                    raise Conflict(
+                        f"Sample size must be less than or equal to: {single_contest.total_ballots_cast} (the total number of ballots in the targeted contest)"
+                    )
+                count += 1
+
+    total_ballots = 0
+    if election.audit_type == AuditType.BALLOT_COMPARISON:
+        contest_sample_size_arr = list(round["sampleSizes"].values())
+        sample_size_count = len(contest_sample_size_arr) - 2
+        for single_contest in election.contests:
+            contest_sample_size = contest_sample_size_arr[sample_size_count - 1]
+            total_ballots = 0
+            if single_contest.is_targeted:
+                for jurisdiction in single_contest.jurisdictions:
+                    total_ballots += (
+                        jurisdiction.manifest_num_batches
+                        if jurisdiction.manifest_num_batches
+                        else 0
+                    )
+                if contest_sample_size > total_ballots:
+                    raise Conflict(
+                        f"Sample size must be less than or equal to: {total_ballots} (the total number of ballots in the targeted contest)"
+                    )
+                sample_size_count -= 1
```

Comment:
for ballot comparison, we can actually use the exact same strategy as ballot polling. `contest.total_ballots_cast` will be populated based on the number of ballots in the CVRs by the time the round gets created.